docs/ballooning.md

```diff
@@ -249,7 +254,31 @@ device has support for the following statistics:
 - `VIRTIO_BALLOON_S_HTLB_PGFAIL`: The number of failed hugetlb page allocations
   in the guest.
 
-The driver is queried for updated statistics every time the amount of time
+Since linux v6.12, following metrics added(omitted in < v6.12):
+
+- `VIRTIO_BALLOON_S_OOM_KILL`: OOM killer invocations, indicating critical
+  memory pressure.
+- `VIRTIO_BALLOON_S_ALLOC_STALL`: Counter of Allocation enter a slow path to
+  gain more memory page. The reclaim/scan metrics can reveal what is actually
+  happening.
+- `VIRTIO_BALLOON_S_ASYNC_SCAN`: Amount of memory scanned asynchronously.
+- `VIRTIO_BALLOON_S_DIRECT_SCAN`: Amount of memory scanned directly.
+- `VIRTIO_BALLOON_S_ASYNC_RECLAIM`: Amount of memory reclaimed asynchronously.
+- `VIRTIO_BALLOON_S_DIRECT_RECLAIM`: Amount of memory reclaimed directly.
+
+When the pages_high watermark is reached, Linux `kswapd` performs asynchronous
+page reclaim, which increases ASYNC_SCAN and ASYNC_RECLAIM.
+
+When a process allocates more memory than the kernel can provide, the process is
+stalled while pages are reclaimed directly, which increases DIRECT_SCAN and
+DIRECT_RECLAIM.
+
+The ratio between DIRECT_RECLAIM/DIRECT_SCAN gets low, the guest OS is thrashing
```

Comment:
```suggestion
When the ratio between DIRECT_RECLAIM/DIRECT_SCAN gets low, the guest OS is thrashing
```

Also do you know at what point would it be considered a low ratio?

Reply:
I've replaced this part to `man sar` ref😄